### PR TITLE
Added Provide-Capability and resolution=optional 

### PR DESCRIPTION
**Pull request by vsingleton**
per suggestions from Ray Auge and Neil Griffin

### CODE REVIEW

File: portlet-api/pom.xml [hunk 1]
```diff
@@ -103,6 +103,21 @@
...
             </plugin>
+            <plugin>
+               <groupId>org.apache.felix</groupId>
+               <artifactId>maven-bundle-plugin</artifactId>
+               <version>3.2.0</version>
+               <extensions>true</extensions>
+               <configuration>
+                  <instructions>
+                     <Import-Package>
+                        javax.enterprise.context;version="[1.1,2)";resolution:=optional,
```
Comment (rotty3000) [inline portlet-api/pom.xml]: I would probably remove the manual package import version here.

File: portlet-api/pom.xml [hunk 2]
```diff
@@ -103,6 +103,21 @@
...
+                        javax.inject;resolution:=optional,*
+                     </Import-Package>
+                     <Provide-Capability>osgi.contract; osgi.contract=JavaPortlet; version:List&lt;Version&gt;="3,2,1"; uses:="javax.enterprise.context,javax.inject,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom"</Provide-Capability>
```
Comment (rotty3000) [inline portlet-api/pom.xml]: The uses here should include the packages exported by the spec API rather than the packages used by the API.

**Review by rotty3000 — CHANGES_REQUESTED** [review]

File: portlet-api/pom.xml (hunk 2 above)
Comment (rotty3000) [reply portlet-api/pom.xml]: To be clear, don't include the packages used by the API... only include the packages provided (exported) by the API.

Comment (rotty3000) [reply portlet-api/pom.xml]: The other packages used by the API which are not part of the API should already be covered by the uses clauses generated by bnd on the Export-Packages directive.

Comment (vsingleton) [reply portlet-api/pom.xml]: I think i have it correct now ... could you check again?  Thanks.

**Comment by rotty3000** [comment]
May I ask you to dump the generated manifest here so I can view it before I give the thumbs up?

**Comment by vsingleton** [comment]
Absolutely.  Here it is:
https://liferayeng.slack.com/files/vernon.singleton/F3EPC03TK/dump_of_the_generated_manifest.txt

**Comment by rotty3000** [comment]
FYI, I'm putting it here for posterity:
```
$ bnd print portlet-api/target/portlet-api-3.0-SNAPSHOT.jar 
[MANIFEST portlet-api-3.0-SNAPSHOT]
Bnd-LastModified                         1481828987586                           
Build-Jdk                                1.7.0_79                                
Built-By                                 apple                                   
Bundle-Description                       The Java Portlet API version 3.0 developed by the Java Community Process JSR-362 Expert Group.
Bundle-DocURL                            http://portals.apache.org/pluto/portlet-api
Bundle-License                           http://www.apache.org/licenses/LICENSE-2.0.txt
Bundle-ManifestVersion                   2                                       
Bundle-Name                              Java Portlet API V3.0                   
Bundle-SymbolicName                      javax.portlet.api                       
Bundle-Vendor                            The Apache Software Foundation          
Bundle-Version                           3.0.0.SNAPSHOT                          
Created-By                               Apache Maven Bundle Plugin              
Export-Package                           javax.portlet;uses:="javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom";version="3.0.0",javax.portlet.annotations;uses:="javax.enterprise.context,javax.inject";version="3.0.0",javax.portlet.filter;uses:="javax.portlet,javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom";version="3.0.0"
Implementation-Title                     Java Portlet API V3.0                   
Implementation-Version                   3.0-SNAPSHOT                            
Import-Package                           javax.enterprise.context;resolution:=optional;version="[1.1,2)",javax.inject;resolution:=optional,javax.portlet,javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom
Manifest-Version                         1.0                                     
Provide-Capability                       osgi.contract;osgi.contract=JavaPortlet;version:List<Version>="3,2,1";uses:="javax.portlet,javax.portlet.annotations,javax.portlet.filter"
Require-Capability                       osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"
Specification-Version                    3.0-SNAPSHOT                            
Tool                                     Bnd-3.2.0.201605172007                  

[IMPEXP]
Import-Package
  javax.enterprise.context               {resolution:=optional, version=[1.1,2)}
  javax.inject                           {resolution:=optional}
  javax.portlet                          
  javax.portlet.annotations              
  javax.servlet                          
  javax.servlet.http                     
  javax.xml.namespace                    
  org.w3c.dom                            
Export-Package
  javax.portlet                          {version=3.0.0}
  javax.portlet.annotations              {version=3.0.0}
  javax.portlet.filter                   {version=3.0.0}
```

**Comment by rotty3000** [comment]
One comment I got back is that it would be friendlier if the bsn (Bundle-SymbolicName) matched the artifactId.

**Comment by vsingleton** [comment]
Great feedback.  Just made the change.  Here is the latest manifest:
```
$ bnd print portlet-api/target/portlet-api-3.0-SNAPSHOT.jar 
[MANIFEST portlet-api-3.0-SNAPSHOT]
Bnd-LastModified                         1481843772468                           
Build-Jdk                                1.7.0_79                                
Built-By                                 apple                                   
Bundle-Description                       The Java Portlet API version 3.0 developed by the Java Community Process JSR-362 Expert Group.
Bundle-DocURL                            http://portals.apache.org/pluto/portlet-api
Bundle-License                           http://www.apache.org/licenses/LICENSE-2.0.txt
Bundle-ManifestVersion                   2                                       
Bundle-Name                              Java Portlet API V3.0                   
Bundle-SymbolicName                      portlet-api                             
Bundle-Vendor                            The Apache Software Foundation          
Bundle-Version                           3.0.0.SNAPSHOT                          
Created-By                               Apache Maven Bundle Plugin              
Export-Package                           javax.portlet;uses:="javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom";version="3.0.0",javax.portlet.annotations;uses:="javax.enterprise.context,javax.inject";version="3.0.0",javax.portlet.filter;uses:="javax.portlet,javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom";version="3.0.0"
Implementation-Title                     Java Portlet API V3.0                   
Implementation-Version                   3.0-SNAPSHOT                            
Import-Package                           javax.enterprise.context;resolution:=optional;version="[1.1,2)",javax.inject;resolution:=optional,javax.portlet,javax.portlet.annotations,javax.servlet,javax.servlet.http,javax.xml.namespace,org.w3c.dom
Manifest-Version                         1.0                                     
Provide-Capability                       osgi.contract;osgi.contract=JavaPortlet;version:List<Version>="3,2,1";uses:="javax.portlet,javax.portlet.annotations,javax.portlet.filter"
Require-Capability                       osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"
Specification-Version                    3.0-SNAPSHOT                            
Tool                                     Bnd-3.2.0.201605172007                  

[IMPEXP]
Import-Package
  javax.enterprise.context               {resolution:=optional, version=[1.1,2)}
  javax.inject                           {resolution:=optional}
  javax.portlet                          
  javax.portlet.annotations              
  javax.servlet                          
  javax.servlet.http                     
  javax.xml.namespace                    
  org.w3c.dom                            
Export-Package
  javax.portlet                          {version=3.0.0}
  javax.portlet.annotations              {version=3.0.0}
  javax.portlet.filter                   {version=3.0.0}```

**Comment by rotty3000** [comment]
Yup, it all looks good!

**Comment by ngriffin7a** [comment]
Merged. Thank you.

**Pull request closed without merging**